### ISSUE TITLE
Additions to times

### ISSUE THREAD

**Issue by jlp765** [nim-lang/Nim#3609]
This is an RFC on some `times` related procs.

I made some helper procs to work with the `TimeInterval` type

```
proc tiMSecs*(ms: int): TimeInterval {.inline.} = initInterval(ms)
  ## TimeInterval of `ms` milliseconds
proc tiSecs*(s: int): TimeInterval {.inline.} = initInterval(0,s)
  ## TimeInterval of `s` seconds
proc tiMins*(m: int): TimeInterval {.inline.} = initInterval(0,0,m)
  ## TimeInterval of `m` minutes
proc tiHrs*(h: int): TimeInterval {.inline.} = initInterval(0,0,0,h)
  ## TimeInterval of `h` hours
proc tiDays*(d: int): TimeInterval {.inline.} = initInterval(0,0,0,0,d)
  ## TimeInterval of `d` days
proc tiMonths*(m: int): TimeInterval {.inline.} = initInterval(0,0,0,0,0,m)
  ## TimeInterval of `m` months
proc tiYrs*(y: int): TimeInterval {.inline.} = initInterval(0,0,0,0,0,0,y)
  ## TimeInterval of `y` years

proc `+=`*(t: var Time, ti: TimeInterval) =
  ## adds the interval `ti` to Time `t`
  ## by converting to localTime, adding the interval, and converting back
  t = timeInfoToTime(getLocalTime(t) + ti)
    
proc `+`*(t: Time, ti: TimeInterval): Time =
  ## adds the interval `ti` to Time `t`
  ## by converting to localTime, adding the interval, and converting back
  result = timeInfoToTime(getLocalTime(t) + ti)
  
proc `-=`*(t: var Time, ti: TimeInterval) =
  ## adds the interval `ti` to Time `t`
  ## by converting to localTime, adding the interval, and converting back
  t = timeInfoToTime(getLocalTime(t) - ti)
  
proc `-`*(t: Time, ti: TimeInterval): Time =
  ## adds the interval `ti` to Time `t`
  ## by converting to localTime, adding the interval, and converting back
  result = timeInfoToTime(getLocalTime(t) - ti)
  
```

which allows
```
var
  t = getTime()
echo getLocalTime(t), "\n", getLocalTime(t + tiSecs(-3))
t += tiHrs(2)
echo getLocalTime(t), "\n", getLocalTime(t) - tiSecs(3)
t -= tiHrs(2)
echo getLocalTime(t)

var ti = getLocalTime(t)
echo ti-tiSecs(3)+tiYrs(1)
```

Should this be a PR and get added to the time module?

**Comment by Araq** [comment]
Yeah. We need to discuss the 'ti*' names, but it's a good feature.

**Comment by jlp765** [comment]
I'm hopeless with good (Nim compliant) naming of variables/types/procs/....

Any suggestions?

**Comment by kirbyfan64** [comment]
```
secondsInterval
millisecondsInterval # or msInterval
etc.
```

**Comment by dom96** [comment]
I would say just call it:

```
miliseconds
seconds
minutes
hours
etc.
```

That way we can write ``2.hours`` or ``50.miliseconds``.

**Comment by jlp765** [comment]
**@dom96** - looks like your suggestion is the way to go.

**Comment by dom96** [comment]
Go for it :)

**Comment by dom96** [comment]
Fixed via #3626.
